compiler/AST/AggregateType.cpp

```diff
@@ -3045,11 +3045,51 @@ void AggregateType::addClassToHierarchy(std::set<AggregateType*>& localSeen) {
 
   globalSeen.insert(this);
 
-  addRootType();
+  // Note: logic in the Dyno scope resolver will sometimes catch multiple inheritance,
+  // but not in every case. So for now, duplicate some checks here.
+  //
+  // TODO: what's a good way to centralize the multiple inheritance handling?
+  Expr* firstParent = nullptr;
 
   // Walk the base class list, and add parents into the class hierarchy.
   for_alist(expr, inherits) {
-    AggregateType* pt = discoverParentAndCheck(expr);
+    AggregateType* pt;
+    InterfaceSymbol* isym;
+    discoverParentAndCheck(expr, pt, isym);
+
+    // Handle class A : MyInterface and continue early.
+    if (isym) {
+      SET_LINENO(expr);
+
+      // For classes, we need to include management style in the implements
+      // statement so that it's picked up when necessary.
+      Symbol* implementFor = this->symbol;
+      if (this->isClass() && isClassLikeOrManaged(this)) {
+        Type* useType =
+          this->getDecoratedClass(ClassTypeDecorator::GENERIC_NONNIL);
+        implementFor = useType->symbol;
+      }
+
+      auto ifcActuals = new CallExpr(PRIM_ACTUALS_LIST, new SymExpr(implementFor));
+      auto istmt = ImplementsStmt::build(isym->name, ifcActuals, nullptr);
+      this->symbol->defPoint->insertAfter(istmt);
+
+      expr->remove();
+      continue;
+    }
+
+    if (this->isRecord()) {
+      USR_FATAL_CONT(expr, "inheritance is not currently supported for records");
+      USR_PRINT("Thoughts on what record inheritance should entail can be added to "
+                "https://github.com/chapel-lang/chapel/issues/6851.");
```

Comment:
This issue is closed & we have no plans to revisit it. I don't think that we should be printing out a link to it in our error messages. Additionally, it appears that we weren't printing it out in practice previously, for whatever reason.